### PR TITLE
feat(to_unixtime): add timestamp types as arguments

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add more argument types for `to_unixtime` function

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link
https://github.com/GreptimeTeam/greptimedb/issues/1103

### CODE REVIEW

**Comment by killme2008** [comment]
Is it still working? We can ship it in 0.3.1 next week.

**Comment by etolbakov** [comment]
Hey Dennis, 
yes, I was catching up with @waynexia a couple of days ago regarding the question that I had.
I'm clear now and will try allocating some time during the weekend.

File: src/common/function/src/scalars/timestamp/to_unixtime.rs [hunk 1]
```diff
@@ -79,6 +91,43 @@ impl Function for ToUnixtimeFunction {
...
                 )))
             }
+            ConcreteDataType::Int64(_) | ConcreteDataType::Int32(_) => {
+                let array = columns[0].to_arrow_array();
+                Ok(Arc::new(Int64Vector::try_from_arrow_array(&array).unwrap()))
+            }
+            ConcreteDataType::Timestamp(ts) => {
+                let array = columns[0].to_arrow_array();
+                let value = match ts {
+                    TimestampType::Second(_) => {
```
Comment (etolbakov) [inline src/common/function/src/scalars/timestamp/to_unixtime.rs:101]: a bit of duplication here..was trying to extract a generic method but could not find one so far. Your suggestions will be highly appreciated 🙏

**Comment by etolbakov** [comment]
@killme2008 @waynexia 
Could you please take a look when you have a spare minute?

**Review by killme2008 — APPROVED** [review]
LGTM

File: tests/cases/standalone/common/select/dummy.sql [hunk 2]
```diff
@@ -11,3 +11,21 @@ select "A";
...
+
+select * from test_unixtime;
+
+select a from test_unixtime;
+
+select b from test_unixtime;
+
+select TO_UNIXTIME(b) from test_unixtime;
+
+DROP TABLE test_unixtime;
```
Comment (waynexia) [inline tests/cases/standalone/common/select/dummy.sql:31]: nit: lack of empty line

**Review by waynexia — APPROVED** [review]
LGTM, thanks @etolbakov 👍

**Comment by etolbakov** [comment]
@killme2008 @waynexia
I've addressed code review suggestions.
Thanks for pointing me out the `paste` crate, I found it useful.
I see that the `test_alter_region_with_reopen` test failed, I've double checked it works locally fine. Could it be the flaky behaviour? Would you mind just restart the action for me please?

**Comment by killme2008** [comment]
> @killme2008 @waynexia I've addressed code review suggestions. Thanks for pointing me out the `paste` crate, I found it useful. I see that the `test_alter_region_with_reopen` test failed, I've double checked it works locally fine. Could it be the flaky behaviour? Would you mind just restart the action for me please?

Great work. Looks like it's a known bug, will be fixed at https://github.com/GreptimeTeam/greptimedb/pull/1759

**Pull request merged**